### PR TITLE
Add code to wrap and unwrap raw essence with Grains

### PR DESCRIPTION
**Pull request by samdbmg**
- Adds `GrainWrapper` helper to wrap raw essence in Grains.
- Adds tools to take raw essence (e.g. from an ffmpeg pipe) and wrap it into GSF.
- Adds a tool to dump GSF file to a file or pipe (e.g to ffmpeg).
- Adds a tool to dump the segments found within a GSF file.

Follows behind #50 - relevant changes are [31902fb..f1221e76](https://github.com/bbc/rd-apmm-python-lib-mediagrains/pull/51/files/31902fb..f1221e76)

Pivotal [#169340688](https://www.pivotaltracker.com/story/show/169340688)

### CODE REVIEW

File: mediagrains/utils/grain_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+from mediatimestamp.immutable import TimeRange
+
+from ..grain import GRAIN
+
+
+class GrainWrapper(object):
+    """Raw input and wrap it in Grains"""
+    def __init__(
+        self,
+        grain_constructor: typing.Callable,
```
Comment (jamesba) [inline mediagrains/utils/grain_wrapper.py]: Should this be better annotated? `typing.Callable` is pretty vague

File: mediagrains/tools/_file_or_pipe.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utility function to open a file or a pipe"""
+import sys
+from contextlib import contextmanager
+
+
+@contextmanager
+def file_or_pipe(file_or_pipe: str, mode: str):
```
[the remark this answers is not in the document]
Comment (samdbmg) [reply mediagrains/tools/_file_or_pipe.py]: Yup, 2b0fd49

File: mediagrains/utils/grain_wrapper.py (hunk 1 above)
Comment (samdbmg) [reply mediagrains/utils/grain_wrapper.py]: I've reworked this based on our discussion to accept a Grain instead. How's 4c4dde7?

**Comment by jamesba** [comment]
LGTM

**Pull request merged**